app/models/plan.rb

```diff
@@ -2,11 +2,25 @@ class Plan < ApplicationRecord
   before_create :generate_token
   validates :name, :price, presence: true
   belongs_to :customer
+  has_many :plan_price_history, dependent: :destroy
+  after_save :save_price_history
+
+  def list_price_history
+    plan_price_history
+      .all
+      .where.not(id: plan_price_history.last)
+      .order('created_at DESC')
```

Comment:
```suggestion
    plan_price_history
      .order(created_at: :desc)
      .offset(1)
```

Como a ordem está `desc` vocês podem dar offset 1 pra não pegar o último, evita uma query mais complexa 😄

Reply:
O `all` é desnecessário porque o has_many já pegaria todos